google/web-stories-wp
# Taxonomies: Add indentation for children of hierarchical terms

## Context
Discovered during bug bash. It's hard when selecting a category as a parent to know how it fits in the nesting.


## Summary
This PR orders the dropdown options such that all children options are below their parent and have an indent next to them to indicate their level.


## Relevant Technical Choices
Initially was working on passing in a custom `renderItem` that was just an extension of the base `DefaultListElement`, however I think I ran into a styled components bug or something because in react dev tools, it was definitely receiving the appended `$level` prop, however the output css would never receive that prop. to mitigate this, I just added a small style to the `DefaultListElement` that indents items if they have a `$level` prop.


## To-do
NA


## User-facing changes
In Editor -> Document -> Taxonomies -> Categories -> Add new Category, the dropdown to select the parent value now has visual indicators to respect the Category hierarchy. Now look like:
<img width="1407" alt="Screen Shot 2021-10-08 at 10 53 44 AM" src="https://user-images.githubusercontent.com/35983235/136587137-9579e6cc-d0b6-472b-8058-9d1b1ac7484f.png">


## Testing Instructions






This PR can be tested by following these steps:

1. Open up the editor -> document tab -> categories panel
2. Make sure you have some nested Categories already (if not create some)
3. Add a new category and select a parent for it
4. See that the dropdown now has visual indicators for the hierarchy, like so:

<img width="1407" alt="Screen Shot 2021-10-08 at 10 53 44 AM" src="https://user-images.githubusercontent.com/35983235/136587626-094263bd-63be-4b39-b601-a2aa3eb57e1e.png">


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #9280
